mkosi/distributions/none.py

```
@@ -0,0 +1,50 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+from collections.abc import Sequence
+
+from mkosi.architecture import Architecture
+from mkosi.distributions import Distribution, DistributionInstaller, PackageType
+from mkosi.log import die
+from mkosi.state import MkosiState
+
+
+class NoneInstaller(DistributionInstaller):
+    @classmethod
+    def filesystem(cls) -> str:
+        return "ext4"
+
+    @classmethod
+    def package_type(cls) -> PackageType:
+        return PackageType.none
+
+    @classmethod
+    def default_release(cls) -> str:
+        return "0"
+
+    @classmethod
+    def default_tools_tree_distribution(cls) -> Distribution:
+        return Distribution.none
+
+    @classmethod
+    def tools_tree_packages(cls) -> list[str]:
+        return []
+
+    @classmethod
+    def setup(cls, state: MkosiState) -> None:
+        pass
+
+    @classmethod
+    def install(cls, state: MkosiState) -> None:
+        pass
+
+    @classmethod
+    def install_packages(cls, state: MkosiState, packages: Sequence[str], apivfs: bool = True) -> None:
+        pass
```

We should fail explicitly here if `packages` is not empty